lib/client/ipam_block_reader_writer.go

```diff
@@ -78,30 +78,25 @@ func (rw blockReaderWriter) claimNewAffineBlock(
 	// If givenPools is not empty, use it.  Otherwise, default to
 	// all configured pools.
 	pools := []cnet.IPNet{}
-	if len(givenPools) != 0 {
-		for _, pool := range givenPools {
-			// Validate the given pool is actually configured and matches the version.
-			if err := rw.checkPoolAvailability(pool); err != nil {
-				return nil, err
-			} else if version.Number != pool.Version() {
-				estr := fmt.Sprintf("The given pool (%s) does not match IP version %d", pool.String(), version.Number)
-				return nil, goerrors.New(estr)
-			}
-			pools = append(pools, pool)
-		}
-	} else {
-		// Default to all configured pools.
-		allPools, err := rw.client.IPPools().List(api.IPPoolMetadata{})
-		if err != nil {
-			log.Errorf("Error reading configured pools: %s", err)
-			return nil, err
-		}
 
-		// Grab all the IP networks in these pools.
-		for _, p := range allPools.Items {
-			// Don't include disabled pools or pools that don't match
-			// the requested IP version.
-			if !p.Spec.Disabled && version.Number == p.Metadata.CIDR.Version() {
+	// Get all the configured pools.
+	allPools, err := rw.client.IPPools().List(api.IPPoolMetadata{})
+	if err != nil {
+		log.Errorf("Error reading configured pools: %s", err)
+		return nil, err
+	}
+
+	// Go through all the IPPools, if givenPools are provided then only add those else add all the pools.
+	for _, p := range allPools.Items {
+		if len(givenPools) != 0 {
+			for _, pool := range givenPools {
+				// Check for the matching IP Pool and add if it's not disabled.
+				if reflect.DeepEqual(p.Metadata.CIDR, pool) && !p.Spec.Disabled {
+					pools = append(pools, pool)
+				}
+			}
+		} else {
+			if !p.Spec.Disabled {
 				pools = append(pools, p.Metadata.CIDR)
```

Comment:
You still need to check version is correct since List will return both versions.  I think this can still be simplified a bit...

When checking if the pool is one of the given pools you can just `continue` on the outer for loop using a labeled for loop.

And then after performing the given pools check you can check the IP version and the Disabled flag as a separate check (so you don't have to repeat yourself)